coconut-rs/src/impls/serde.rs

```diff
@@ -0,0 +1,59 @@
+use crate::elgamal::PrivateKey;
+use crate::scheme::SecretKey;
+use crate::{Base58, BlindSignRequest, BlindedSignature, PublicKey, Signature, VerificationKey};
+use serde::de::Unexpected;
+use serde::{de::Error, de::Visitor, Deserialize, Deserializer, Serialize, Serializer};
+use std::fmt;
+
+macro_rules! impl_serde {
+    ($struct:ident) => {
+        gensym::gensym! { _impl_serde!{ $struct } }
```

Comment:
Can you explain a bit what's the point of the gensym macro here? I'm not really an expert with macros and don't quite understand the point of UUIDs here

Reply:
> Can you explain a bit what's the point of the gensym macro here? I'm not really an expert with macros and don't quite understand the point of UUIDs here

In order to implement `Deserialize` we need a struct that implements [Visitor](https://docs.serde.rs/serde/de/trait.Visitor.html). As we don't care what the visitor struct will be called `gensym`macro generates a uuid identifier for a struct, we then implement Visitor for it. In the end we get a macro with one parameter instead of two :)

Reply:
So maybe a suggestion, could we just pass an additional argument to the macro which would be the name of the struct? I think I've seen this pattern before in some other library. (so for `Theta` it would be `Theta` or whatever) - this way we'd prevent pulling an additional dependency which I'd think would be a good thing in a crypto library.

Reply:
Love it, security does trump convenience :)